### DISCUSSION TITLE
How to do continuous build in Koka compiler?

### DISCUSSION THREAD

**Discussion by swarnpriya** [koka-lang/koka#475] [General]
I made some changes in the compiler and built the compiler using "stack update; stack build; stack exec koka ..." to test my new changes. Now when I make other changes to the same function (newly added) in the compiler, the built does not incorporate it and gets stuck to the old built (even after performing "stack clean; stack update; stack build; stack exec koka ..." again).

**Comment by TimWhiting** [comment]
You do not need to do `stack clean` / `stack update` every time. In particular I typically use `stack run koka` instead of `stack build` `stack exec koka`. First try doing that.

If that doesn't work:
I can think of a few potential reasons for your issues currently. 
1. You are not running `stack exec` and are just running `koka` and you are getting some other `koka` installation - most likely not this, but it doesn't hurt to check.
2. It could also be a difference between `stack build --fast` and `stack build`, since those create two different optimization levels of the binary. To debug stack issues try running `stack path`, and look for the install path (you can run Koka directly from there). 
3. Another issue is that you might have errors in `stack build` - there is no partial compilation in Haskell, so if there are any type errors or anything, you will not get a new binary. 
4. The final issue that I can think of is that you are getting the new function, but it just isn't doing what you expect due to Haskell's laziness.
